### PR TITLE
Add Template and Action classes and tests

### PR DESCRIPTION
**Pull request by mvriel**

### CODE REVIEW

File: src/phpDocumentor/Renderer/ActionHandlerLocator.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+use Interop\Container\ContainerInterface;
+
+/**
+ * Service that attempts to locate an ActionHandler object based on the class name of a given Action.
+ */
+class ActionHandlerLocator
+{
+    /**
+     * @var ContainerInterface
```
Comment (jaapio) [inline src/phpDocumentor/Renderer/ActionHandlerLocator.php]: missing a summary here

File: tests/unit/phpDocumentor/Renderer/ActionHandlerLocatorTest.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+use Mockery as m;
+
+/**
+ * Tests the functionality for the ActionHandlerLocator class.
+ */
+class ActionHandlerLocatorTest extends \PHPUnit_Framework_TestCase
+{
+    public function testIfLocatorRetrievesAnActionWithHanderAppendedToClassName()
+    {
+        $action = m::mock('phpDocumentor\Renderer\Action');
```
Comment (jaapio) [inline tests/unit/phpDocumentor/Renderer/ActionHandlerLocatorTest.php]: I would use the ::class constant here, Helps when you refactor the name of a class

File: src/phpDocumentor/Renderer/Template.php [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+ * the Template also determines at which location to create pre-rendered output and has to copy
+ * assets such as CSS files, Javascript and images.
+ *
+ * In order to do this you can define a series of Actions with a template that prescribe which
+ * Actions to execute in the specified order.
+ *
+ * With a template you can provide a series of parameters that can be passed to every Action
+ * so that a default piece of information is provided, such as the target location.
+ */
+class Template
```
Comment (jaapio) [inline src/phpDocumentor/Renderer/Template.php]: shouldn't this be final?

**Comment by mvriel** [comment]
@jaapio thanks for the review; I have fixed your feedback and added the missing test for the Template class (had forgotten to commit that in the initial commit *facepalm*)

**Comment by jaapio** [comment]
looks good, Wercker should be able to pass? think you need a rebase here. But can be merged without that.

**Pull request merged**